### PR TITLE
add hvdc category for expert filters

### PR DESCRIPTION
**Pull request by EtienneLt**
linked to https://github.com/gridsuite/commons-ui/pull/524 and https://github.com/gridsuite/gridexplore-app/pull/473

### CODE REVIEW

File: src/main/java/org/gridsuite/filter/utils/expertfilter/ExpertFilterUtils.java [hunk 1]
```diff
@@ -51,11 +51,37 @@ public static <I extends Identifiable<I>> String getFieldValue(FieldType field,
...
                 default -> throw new PowsyblException(TYPE_NOT_IMPLEMENTED + " [" + identifiable.getType() + "]");
             };
         };
     }
 
+    private static String getHvdcLineFieldValue(FieldType field, String propertyName, HvdcLine hvdcLine) {
+        return switch (field) {
+            case CONNECTED_1 -> getTerminalFieldValue(field, hvdcLine.getConverterStation1().getTerminal());
+            case CONNECTED_2 -> getTerminalFieldValue(field, hvdcLine.getConverterStation2().getTerminal());
+            case CONVERTERS_MODE -> String.valueOf(hvdcLine.getConvertersMode());
```
Comment (thangqp) [inline src/main/java/org/gridsuite/filter/utils/expertfilter/ExpertFilterUtils.java:64]:
```suggestion
            case CONVERTERS_MODE -> hvdcLine.getConvertersMode() != null ? hvdcLine.getConvertersMode().name() : null;
```

File: src/test/java/org/gridsuite/filter/expertfilter/NumberExpertRuleTest.java [hunk 2]
```diff
@@ -2383,4 +2388,204 @@ private static Stream<Arguments> provideArgumentsForStaticVarCompensatorTest() {
...
+        Terminal terminal = Mockito.mock(Terminal.class);
+        VoltageLevel voltageLevel = Mockito.mock(VoltageLevel.class);
+        Mockito.when(voltageLevel.getNominalV()).thenReturn(13.0);
+        Mockito.when(terminal.getVoltageLevel()).thenReturn(voltageLevel);
+
+        // Converter Stations fields
+        HvdcConverterStation converterStation = Mockito.mock(HvdcConverterStation.class);
+        Mockito.when(converterStation.getTerminal()).thenReturn(terminal);
+        Mockito.when(hvdcLine.getConverterStation1()).thenReturn(converterStation);
+        Mockito.when(hvdcLine.getConverterStation2()).thenReturn(converterStation);
```
Comment (thangqp) [inline src/test/java/org/gridsuite/filter/expertfilter/NumberExpertRuleTest.java:2408]: In a real network, is it possible a hdvc line share the same converterStation and terminal?
If not it is better to separate in two stations so two terminals

File: src/test/java/org/gridsuite/filter/expertfilter/NumberExpertRuleTest.java [hunk 3]
```diff
@@ -2383,4 +2388,204 @@ private static Stream<Arguments> provideArgumentsForStaticVarCompensatorTest() {
         );
     }
 
+    private static Stream<Arguments> provideArgumentsForHvdcLinesTest() {
```
Comment (thangqp) [inline src/test/java/org/gridsuite/filter/expertfilter/NumberExpertRuleTest.java:2391]: I don't see the test for nominalV of hdvc line, that's why test coverage is not 100%

File: src/test/java/org/gridsuite/filter/expertfilter/NumberExpertRuleTest.java (hunk 2 above)
Comment (EtienneLt) [reply src/test/java/org/gridsuite/filter/expertfilter/NumberExpertRuleTest.java:2409]: ok I'll change that

File: src/main/java/org/gridsuite/filter/utils/expertfilter/ExpertFilterUtils.java [hunk 4]
```diff
@@ -51,11 +51,40 @@ public static <I extends Identifiable<I>> String getFieldValue(FieldType field,
...
+            case ACTIVE_POWER_SET_POINT -> String.valueOf(hvdcLine.getActivePowerSetpoint());
+            case MAX_P -> String.valueOf(hvdcLine.getMaxP());
+            case DC_NOMINAL_VOLTAGE -> String.valueOf(hvdcLine.getNominalV());
+            case CONVERTER_STATION_ID_1 -> hvdcLine.getConverterStation1().getId();
+            case CONVERTER_STATION_NOMINAL_VOLTAGE_1 ->
+                String.valueOf(hvdcLine.getConverterStation1().getTerminal().getVoltageLevel().getNominalV());
+            case CONVERTER_STATION_ID_2 -> hvdcLine.getConverterStation2().getId();
+            case CONVERTER_STATION_NOMINAL_VOLTAGE_2 ->
+                String.valueOf(hvdcLine.getConverterStation2().getTerminal().getVoltageLevel().getNominalV());
+            case NOMINAL_VOLTAGE -> String.valueOf(hvdcLine.getNominalV());
```
Comment (thangqp) [inline src/main/java/org/gridsuite/filter/utils/expertfilter/ExpertFilterUtils.java:74]: Clean this line

**Review by thangqp — APPROVED** [review]

**Pull request merged**